packages/components/src/Heading/Heading.tsx

```diff
@@ -0,0 +1,44 @@
+import { TextProps, TypographyComponent } from "../Text/Text";
+import React from "react";
+import { styled } from "twin.macro";
+
+type HeadingElement = "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
+
+export type HeadingProps = TextProps & {
+  /**
+   * This prop can be used to specify which size heading should
+   * actually be rendered, in the case that you want to render an element
+   * as one heading but style it as if it were another. If both an `as` prop
+   * and a `size` prop are passed, the `as` will be used to determine the element
+   * and the `size` will be used to determine the styling.
+   */
+  as?: HeadingElement;
+};
+
+const HeadingComponent = styled(TypographyComponent).attrs({
+  bold: true,
+})``;
```

Comment:
Currently in traject, all headings are indeed bold. so I'm comfortable enforcing this, and we can revisit if there's a design need later

also this is how the component is rendered in DOM:
<img width="600" alt="Screen Shot 2021-02-05 at 12 15 11 PM" src="https://user-images.githubusercontent.com/15840841/107070467-67022c00-67b1-11eb-9cb7-7ff29d114ce5.png">

Reply:
also this is how the component is rendered in DOM:
<img width="600" alt="Screen Shot 2021-02-05 at 12 15 11 PM" src="https://user-images.githubusercontent.com/15840841/107070467-67022c00-67b1-11eb-9cb7-7ff29d114ce5.png">

versus directly using `TypographyComponent`:
<img width="600" alt="Screen Shot 2021-02-05 at 1 19 28 PM" src="https://user-images.githubusercontent.com/15840841/107073133-f6f5a500-67b4-11eb-9bb5-dedbffc7d3b0.png">

I think the extra className is ok? this follows [styled-components' guidance](https://styled-components.com/docs/faqs#when-to-use-attrs) that
> The rule of thumb is to use attrs when you want every instance of a styled component to have that prop, and pass props directly when every instance needs a different one

Reply:
🤔 Do we want headings to always be bold in the design system? Or are we going for maxmizing the ease of integrating into Learning Platform currently?

Reply:
Also, I wouldn't worry about the extra class.

Reply:
ehhhh we kinda went back and forth...Sean originally wanted always bold, but then some LP designers wanted a non-bold option, but in LP they're all bold anyways...though since Along allows passing a weight maybe we should keep it flexible

Reply:
I added the option back for now ~ flexibility ~